comprl/server/hockeygame.py

```diff
@@ -37,6 +37,11 @@ def __init__(self, players: list[IPlayer]) -> None:
         # Bool if all rounds are finished
         self.finished = False
 
+        # array storing all actions to be saved later.
+        # The array contains one subarray per round.
+        self.actions = np.array([])
+        self.actions_this_round = np.array([])
```

Comment:
I think it would be better to implement the `self.actions` in `IGame` and fill it in the `_game_cycle`. It is possible to save the actions for any game in the same way, because they are all stored in float arrays, so it is easier to implement this for all games in `IGame` and not for every game separately. 
Instead of the `self.actions_this_round` array, you could just use the `self.current_actions` list.

Reply:
I agree, implementing this in `IGame` is generally better. The reason I didn't do this and used `self.actions_this_round` is, because every other round played the actions are swapped, as the players switch sides. 
As this is a special case, only relevant in `HockeyGame`, if we decide to move this to `IGame`, the actions are stored without swapping. But maybe this is even better, as then you don't get confused, about which action is from whom.

Time is a bit short for me at the moment, but I will try to change it in the next days.